src/components/OpenWith/OpenWithButton.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @flow
+ * @file Preview Open With button
+ * @author Box
+ */
+
+import * as React from 'react';
+import Button from 'box-react-ui/lib/components/button/Button';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import WithTooltip from './WithTooltip';
+import ICON_FILE_MAP from './IconFileMap';
+import messages from '../messages';
+
+type Props = {
+    extension: ?string,
+    displayIntegration: ?Integration,
+    onClick: ?Function,
+    tooltipText?: string | Object,
+    isDisabled: boolean,
+    isLoading: boolean,
+    icon?: string,
+};
+
+const INTEGRATION_ICON_CLASS = 'integration-icon';
+
+const OpenWithButton = ({
+    extension,
+    onClick,
+    displayIntegration,
+    tooltipText,
+    isDisabled = false,
+    isLoading = true,
+}: Props) => {
+    const { displayName = null, appIntegrationId = null } =
+        displayIntegration || {};
+    let DisplayIcon = ICON_FILE_MAP[displayName || 'default'];
+
+    let iconProps = {
```

Comment:
Would love some feedback on how to handle this more cleanly. The issue here is that depending on what component we use FileIcon vs. a special integration icon e.g. IconAdobeSign, the props will be different.

height, width vs. dimension.

Also, the specific integration icons have a different viewbox, and require extra padding.

Reply:
I think the simple solution would be to handle "default" separately:

```
const IntegrationIcon = displayName && ICON_FILE_MAP[displayName];

...(in render)
{ IntegrationIcon ? <IntegrationIcon height={26} width={26} /> : <FileIcon dimension={26} /> }
```